java-frontend/src/main/java/org/sonar/plugins/java/api/semantic/Symbol.java

```diff
@@ -183,14 +183,24 @@
     List<Type> thrownTypes();
 
     /**
-     * Retrieve the overridden symbol, which may may not be able to determine (returning 'unknown' symbol).
-     * Note that if the method returns null, the method is not overriding any method for sure.
+     * Retrieve the overridden symbol, which might not be able to determined (returning 'unknown' symbol).
      *
      * @return the overridden symbol, null if the method is not overriding any method or overriding can not be determined (incomplete semantic)
+     * @deprecated This method only returns the first symbol that has been overridden, rather than all symbols that are being overridden.
+     *             Use {@link #overriddenSymbols()} to get a list of all symbols
      */
     @Nullable
+    @Deprecated
     Symbol.MethodSymbol overriddenSymbol();
 
+    /**
+     * Retrieve the overridden symbols in all the known type hierarchy.
+     * Note that it will only returns the symbols which might be determined from known types. The list might therefore not be complete in case of missing dependencies.
+     *
+     * @return the overridden symbols, or an empty list if the method is not overriding any method or overriding can not be determined (incomplete semantics)
+     */
+    List<Symbol.MethodSymbol> overriddenSymbols();
```

Comment:
We changed the public API, we should document it in the API changes [documentation](https://github.com/SonarSource/sonar-java/blob/master/sonar-java-plugin/src/main/resources/static/documentation.md).

Reply:
indeed. I will document it.